authorization/src/main/kotlin/org/modelix/authorization/PermissionManagementPage.kt

```diff
@@ -0,0 +1,254 @@
+    route("permissions") {
+        get("manage") {
+            call.respondHtml {
+                buildPermissionManagementPage(call, application.plugin(ModelixAuthorization))
+            }
+        }
+        post("grant") {
+            val formParameters = call.receiveParameters()
+            val userId = formParameters["userId"]
+            val roleId = formParameters["roleId"]
+            require(userId != null || roleId != null) { "userId or roleId required" }
+            val permissionId = requireNotNull(formParameters["permissionId"]) { "permissionId not specified" }
+
+            // a user can grant his own permission to other users
+            checkPermission(PermissionParts.fromString(permissionId))
+
+            if (userId != null) {
+                application.plugin(ModelixAuthorization).config.accessControlPersistence.update {
+                    it.withGrantToUser(userId, permissionId)
+                }
+            }
+            if (roleId != null) {
+                application.plugin(ModelixAuthorization).config.accessControlPersistence.update {
+                    it.withGrantToRole(roleId, permissionId)
+                }
+            }
+            call.respond("Granted $permissionId to ${userId ?: roleId}")
+        }
+        post("remove-grant") {
+            call.checkPermission(PermissionSchemaBase.permissionData.write)
+            val formParameters = call.receiveParameters()
+            val userId = formParameters["userId"]
+            val roleId = formParameters["roleId"]
+            require(userId != null || roleId != null) { "userId or roleId required" }
+            val permissionId = requireNotNull(formParameters["permissionId"]) { "permissionId not specified" }
+            if (userId != null) {
+                application.plugin(ModelixAuthorization).config.accessControlPersistence.update {
+                    it.withoutGrantToUser(userId, permissionId)
+                }
+            }
+            if (roleId != null) {
+                application.plugin(ModelixAuthorization).config.accessControlPersistence.update {
+                    it.withoutGrantToUser(roleId, permissionId)
+                }
+            }
+            call.respond("Removed $permissionId to ${userId ?: roleId}")
+        }
+    }
+}
+
+fun HTML.buildPermissionManagementPage(call: ApplicationCall, pluginInstance: ModelixAuthorizationPluginInstance) {
+    val schemaInstance = pluginInstance.config.permissionSchema
+    head {
+        style {
+            //language=CSS
+            +"""
+                table {
+                    border: 1px solid #ccc;
+                    border-collapse: collapse;
+                }
+                td, th {
+                    border: 1px solid #ccc;
+                    padding: 3px 12px;
+                }
+            """.trimIndent()
+        }
+    }
+    body {
+        h1 {
+            +"Grant Permission"
+        }
+        postForm(action = "grant") {
```

Comment:
A second form to grant permissions for `roleId` might be useful.

Reply:
Added